### PR TITLE
feat(model-apps): add headless app-shell mode (spec validation + build wiring + fixture)

### PR DESCRIPTION
**Pull request by samrat-gavale**
## Summary
Introduces a first-class **headless** mode for the model-apps maker: an app declared only from a solution + entities + appShell, with no forms/views/charts/dashboards/pages/commands/webResources. Adds spec validation, mutual-exclusion linting, headless phase wiring in the build pipeline, a canonical fixture, and docs.

## Acceptance criteria coverage
| AC | Unit | Test class | Status |
|---|---|---|---|
| App Spec supports declaring `headless: true` with only solution + entities + appShell | U1 | `HeadlessSpecValidatorTests` | ✅ |
| Spec containing headless=true alongside forbidden surfaces (forms/views/charts/dashboards/pages/commands/webResources) is rejected by lint with clear message | U1 | `HeadlessMutualExclusionLintTests` | ✅ |
| Build pipeline recognizes headless mode and short-circuits non-applicable phases while running the allowlisted phases end-to-end | U2 | `HeadlessBuildWiringTests` | ✅ |
| A canonical headless fixture is included and used by tests to lock the download/upload/edit round-trip shape | U2 | `HeadlessFixtureRoundTripTests` | ✅ |

## Derived goals
- App Spec supports a first-class headless mode declaring an app built from solution + entities + appShell only (no forms/views/charts/dashboards/pages/commands/webResources)
- The lint layer rejects headless=true alongside any forbidden surface with a clear, testable message
- The build pipeline honors headless by allowlisting only the applicable phases; other phases short-circuit deterministically
- A canonical headless fixture exists so tests lock the round-trip shape and future changes don't silently drift the spec

## Units delivered
- **U1** — Headless spec validation + mutual-exclusion lint: new validator + lint rule + rejection with actionable message
- **U2** — Headless build wiring + canonical fixture + docs: phase allowlist wiring, fixture, and roadmap/CHANGELOG updates

## Test results
- Build: clean
- Tests: passing (unit + edge cases)
- Static analysis: clean

## Related work items
- AB#38732772 (parent — headless model-app shell)
- AB#38733417 (U1)
- AB#38733418 (LT1 — PR)

### CODE REVIEW

File: plugins/model-apps/scripts/lib/sdk-build.js [hunk 1]
```diff
@@ -145,6 +145,33 @@ function resolvePhases({ only, skip, from, to } = {}) {
   return active.filter((p) => (!onlySet || onlySet.has(p)) && (!skipSet || !skipSet.has(p)));
 }
 
+// Pure allow-list for a headless model app (spec.headless === true): the reduced phase set
```
Comment (akshaymaloo) [inline plugins/model-apps/scripts/lib/sdk-build.js:148]: rebase with latest branch, most of these files dont exist anymore

File: plugins/model-apps/skills/model-app-maker/SKILL.md [hunk 2]
```diff
@@ -43,6 +43,12 @@ prod-ready** app; don't under-build (a bare table list) or over-build (surfaces
 - **Surfaces** — **generative pages** (modern dashboards / overviews / analytics / landing — the default),
   classic dashboards (opt-in), external URLs
 - **App shell** — the app module + sitemap, with per-subarea icons
+- **Headless (opt-in)** — a **table + sitemap only** shell (no forms/views/charts/dashboards);
```
Comment (akshaymaloo) [inline plugins/model-apps/skills/model-app-maker/SKILL.md]: I would recommend moving this and any other references for headless to a new file under `/references`

In the skill we just say read the file when headless is mentioned in the user prompts ... something like lazy loading

File: plugins/model-apps/scripts/lib/sdk-build.js (hunk 1 above)
Comment (samrat-gavale) [reply plugins/model-apps/scripts/lib/sdk-build.js:148]: Done — rebased onto the latest `users/akmaloo/model-app-authoring` (now `0e18cbf`). The stale files are gone and the PR is now MERGEABLE/CLEAN. The skill rename (`model-app-maker` -> `app-builder`) that landed on the base after this branch was cut is now incorporated. All 404 unit tests pass.

File: plugins/model-apps/skills/model-app-maker/SKILL.md (hunk 2 above)
Comment (samrat-gavale) [reply plugins/model-apps/skills/model-app-maker/SKILL.md]: Done — moved all headless documentation into a new `plugins/model-apps/references/headless-apps.md`. `SKILL.md` and `app-spec-schema.md` now carry only short pointers, and the skill instruction is lazy-load: read `references/headless-apps.md` when the user mentions headless. Also added a CHANGELOG entry.
